python/packages/autogen-core/docs/src/agentchat-user-guide/quickstart.md

```diff
@@ -0,0 +1,84 @@
+---
+
+# Quick Start
+
+AgentChat API, introduced in AutoGen 0.4x, offers a similar level of abstraction as the default Agent classes in AutoGen 0.2x. This guide demonstrates how to migrate basic examples from AutoGen 0.2x to AgentChat in AutoGen 0.4x.
+
+## Installation
+
+Install the `autogen-agentchat` package using pip:
+
+```bash
+
+pip install autogen-agentchat==0.4.0dev0
+```
+
+:::{note}
+For further installation instructions, please refer to the [package information](pkg-info-autogen-agentchat).
+:::
+
+## Migrating from AutoGen 0.2x to 0.4x
+
+The following example illustrates creating a simple agent team with two agents and compares how this is done in AutoGen 0.2x and 0.4x.
+
+1. An `AssistantAgent` (0.2x) / `CodingAssistantAgent` (0.4x) that generates responses using an LLM model.
+2. A `UserProxyAgent` (0.2x) / `CodeExecutorAgent` (0.4x) that executes code snippets and returns the output.
+
+The task is to "Create a plot of NVIDIA and TESLA stock returns YTD from 2024-01-01 and save it to 'nvidia_tesla_2024_ytd.png'."
+
+``````{tab-set}
+
+`````{tab-item} AgentChat (v0.4x)
+```python
+from autogen_agentchat.agents import CodeExecutorAgent, CodingAssistantAgent
+from autogen_agentchat.teams.group_chat import RoundRobinGroupChat
+from autogen_core.components.code_executor import DockerCommandLineCodeExecutor
+from autogen_core.components.models import OpenAIChatCompletionClient
+
+async with DockerCommandLineCodeExecutor(work_dir="coding") as code_executor:
+    code_executor_agent = CodeExecutorAgent("code_executor", code_executor=code_executor)
+    coding_assistant_agent = CodingAssistantAgent(
+        "coding_assistant", model_client=OpenAIChatCompletionClient(model="gpt-4")
+    )
+    group_chat = RoundRobinGroupChat([coding_assistant_agent, code_executor_agent])
+    result = await group_chat.run(
+        task="Create a plot of NVIDIA and TESLA stock returns YTD from 2024-01-01 and save it to 'nvidia_tesla_2024_ytd.png'."
+    )
+    print(result)
+```
+`````
+
+`````{tab-item} v0.2x
+```python
+from autogen import AssistantAgent, UserProxyAgent, config_list_from_json
+
+config_list = config_list_from_json(env_or_file="OAI_CONFIG_LIST")
+assistant = AssistantAgent("assistant", llm_config={"config_list": config_list})
+code_executor_agent = UserProxyAgent(
+    "code_executor_agent",
+    code_execution_config={"work_dir": "coding", "use_docker": True}
+)
+code_executor_agent.initiate_chat(
+    assistant,
+    message="Create a plot of NVIDIA and TESLA stock returns YTD from 2024-01-01 and save it to 'nvidia_tesla_2024_ytd.png'."
+)
+```
+`````
+
+``````
+
+AgentChat in v0.4x provides similar abstractions to the default agents in v0.2x:
+
+- `CodingAssistantAgent` (v0.4x) ~ `AssistantAgent` (v0.2x)
+- `CodeExecutorAgent` (v0.4x) ~ `UserProxyAgent` with code execution (v0.2x)
+
+Key differences:
+
+1. In v0.4x, agent interactions are managed by `Teams` (e.g., `RoundRobinGroupChat`), replacing direct chat initiation.
```

Comment:
I find it confusing that we use the term Team in text but the code uses GroupChat. Is there a plan to extend the Team beyond group chats? if so this text should clarify that and instead say `Teams` (e.g., implementation of group chat-based teams such as ...)

Reply:
Yeah, I had the same thought while writing this ... I think it just might be reasonable to rename it to Team. 

Some other related thoughts. 

- Single `AssistantAgent` that can take params that govern behaviours e.g., `description`, `system_message`, `code_executor`, `tools_schema` ...  
- Single `Team` or `AgentTeam` that takes params that govern behaviours e.g. `max_turns`, `speaker_selection` (round_robin, llm, graph etc).

Reply:
Are there any teams that aren't considered group chats?
If yes, then group chat is more specific and accurate.
If no, renaming to Teams is okay with me.

Reply:
I'd like to chat a bit more about how tools are incorporated. We should have a design session with the (human) team on this.

Reply:
The resolution on tools is in #3665